VIMLoader.ts

```diff
@@ -257,335 +191,126 @@ export class VIMLoader {
-  constructG3D (bfast) {
+  constructG3D (bfast: BFast): G3D.G3d {
     console.log('Constructing G3D')
 
     if (bfast.buffers.length < 2) {
       throw new Error('G3D requires at least two BFast buffers')
     }
 
-    // This will just contain some JSON
+    // Parse first buffer as Meta
     const metaBuffer = bfast.buffers[0]
     if (bfast.names[0] !== 'meta') {
       throw new Error(
         "First G3D buffer must be named 'meta', but was named: " +
           bfast.names[0]
       )
     }
+    const meta = new TextDecoder('utf-8').decode(metaBuffer)
 
-    // Extract each descriptor
-    const attributes = []
+    // Parse remaining buffers as Attributes
+    const attributes: G3D.Attribute[] = []
     const nDescriptors = bfast.buffers.length - 1
     for (let i = 0; i < nDescriptors; ++i) {
-      const desc = bfast.names[i + 1].split(':')
-      if (desc[0].toLowerCase() !== 'g3d' || desc.length !== 6) {
-        throw new Error(
-          "Not a valid attribute descriptor, must have 6 components delimited by ':' and starting with 'g3d' " +
-            desc
-        )
-      }
-      const attribute = {
-        name: desc,
-        association: desc[1], // Indicates the part of the geometry that this attribute is associated with
-        semantic: desc[2], // the role of the attribute
-        attributeTypeIndex: desc[3], // each attribute type should have it's own index ( you can have uv0, uv1, etc. )
-        dataType: desc[4], // the type of individual values (e.g. int32, float64)
-        dataArity: desc[5], // how many values associated with each element (e.g. UVs might be 2, geometry might be 3, quaternions 4, matrices 9 or 16)
-        rawData: bfast.buffers[i + 1], // the raw data (a UInt8Array)
-        data: undefined
-      }
-      attribute.data = this.attributeToTypedArray(attribute)
-      console.log('Attribute ' + i + ' = ' + desc)
+      const attribute = G3D.Attribute.fromString(
+        bfast.names[i + 1],
+        bfast.buffers[i + 1]
+      )
       attributes.push(attribute)
+      console.log(`Attribute ${i} = ${attribute.descriptor.description}`)
     }
 
-    return {
-      attributes: attributes,
-      meta: new TextDecoder('utf-8').decode(metaBuffer)
-    }
-  }
-
-  // Finds the first attribute that has the matching fields passing null matches a field to all
-  findAttribute (VIM, assoc, semantic, index, dataType, arity) {
-    const r = []
-    for (let i = 0; i < VIM.attributes.length; ++i) {
-      const attr = VIM.attributes[i]
-      if (
-        (attr.association === assoc || assoc == null) &&
-        (attr.semantic === semantic || semantic == null) &&
-        (attr.attributeTypeIndex === index || index == null) &&
-        (attr.dataArity === arity || arity == null) &&
-        (attr.dataType === dataType || dataType == null)
-      ) {
-        r.push(attr)
-      }
-    }
-    return r.length > 0 ? r[0] : null
+    return new G3D.G3d(meta, attributes)
```

Comment:
Minor: capitalization inconsistency between "G3D" and "G3d" in `G3D.G3d`. Looking at similar names in the codebase, we seem to align on the lowercase-d form.